modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -42,6 +45,49 @@
     PartitionUnionType = Any
 
 
+def _ray_from_modin_non_partition(api_layer_object: DataFrame | Series):
+    """
+    HACK:
+    This is a hack that makes ray.data.from_modin work with modin dataframe not backed by partitions.
+
+    See also: https://docs.ray.io/en/latest/data/api/doc/ray.data.from_modin.html#ray-data-from-modin for ray.data.from_modin's documentation.
+
+    Parameters
+    ----------
+
+    api_layer_object : DataFrame or Series
+        The API layer object (backed by datawarehouses).
+
+    Returns
+    -------
+
+    A list of Ray.ObjectRef that can be combined into the original ``api_layer_object``
+    """
+
+    import pandas as pd
+    import ray
+    from ray import remote
+
+    from modin.core.storage_formats.pandas.utils import compute_chunksize
+
+    num_cpus = ray.cluster_resources()["CPU"]
+    length = len(api_layer_object)
+    part_size: int = compute_chunksize(length, num_cpus)  # type: ignore
+
+    # Calculate the actual number of partitions.
+    # This might be fewer than length / CPUs because there's a minimum chunk size.
+    num_parts = length // part_size
+
+    @remote
+    def store_range(api_layer_object: DataFrame | Series, part_idx: int):
+        start_idx = part_idx * part_size
+        end_idx = (part_idx + 1) * part_size
+        return pd.DataFrame(api_layer_object.iloc[start_idx:end_idx])
```

Comment:
this feels super-weird - it's going to store the whole object in the store to only extract its piece remotely...

Reply:
BTW @rentruewang did you test that this works on non-pandas storage format (like HDK or some SQL)?

Reply:
Sorry that is not what this commit is for, it's for modin instances that do not have a partition manager.